blog/2024-12-06-deadlock-gadget-lfx-mentorship.md

```diff
@@ -0,0 +1,144 @@
+---
+authors: [snehil-shah]
+description: "Introducing a new gadget for detecting deadlocks"
+draft: false
+tags: ["eBPF", "ig", "Inspektor Gadget", "Gadget", "WASM", "deadlock", "LFX Mentorship"]
+title: "Introducing a new gadget for detecting deadlocks"
+slug: /2024/12/deadlock-gadget-lfx-mentorship
+image: /media/2024-05-31-cncf-mentoring-logo.jpg
+---
+
+Hey! I am [Snehil Shah](https://github.com/Snehil-Shah), a newer inspektor in the community. Over these past few months, I have been working under the mentorship of [Alban Crequy](https://github.com/alban) and [Burak Ok](https://github.com/burak-ok) as part of the [LFX Mentorship](https://mentorship.lfx.linuxfoundation.org/project/7fdda09c-0eb8-466b-9fdf-e4b3c6a1d5b3) program.
+Consider this blog a report on my work during the mentorship.
+
+<!-- truncate -->
+
+## What have I been up to?
+
+My project involved developing a [new gadget](https://github.com/inspektor-gadget/inspektor-gadget/issues/3194) for detecting potential deadlocks in multi-threaded user space applications, using eBPF and WASM.
+
+Debugging and preventing deadlocks in modern, complex codebases can be challenging, and this gadget simplifies the process by identifying risks early, promoting safer and more efficient resource management.
```

Comment:
I think splitting this sentence into two sentences would make it easier for users to follow.

```suggestion
Debugging and preventing deadlocks in modern, complex codebases can be challenging. This gadget simplifies the process by identifying risks early, promoting safer and more efficient resource management.
```